### DISCUSSION TITLE
Need Help with 2FA TOTP Authentication Not Working

### DISCUSSION THREAD

**Discussion by nurfarah28** [community/community#128583] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello GitHub Community,

I am experiencing an issue with two-factor authentication on my GitHub account and could use some guidance. Despite correctly setting up 2FA, I am unable to log in because the TOTP codes generated by my authenticator apps are not being recognized by GitHub. Additionally, when I try to use my recovery codes during the login process, I receive a message that these are not appropriate for the step.

Here are some details about my problem:

Issue: TOTP codes from multiple authenticator apps (including Microsoft Authenticator and Google Authenticator) are consistently rejected.
Attempted Solutions: I have tried re-syncing the time on my devices, reinstalling the apps, and manually entering the setup keys, but none of these steps have resolved the issue.
Recovery Codes: When I use a recovery code, I am informed that a TOTP code is required instead, blocking my access.
This situation is quite puzzling as I am following all standard procedures for 2FA. I'm wondering if anyone here has encountered a similar issue or could offer advice on resolving this? Any insights or suggestions would be greatly appreciated.

Thank you in advance for your help!
![Screenshot 2024-06-14 214517](https://github.com/community/community/assets/172844317/192cdadb-08ca-43b4-a9bc-94893a9ae274)

**Comment by promiselb** [comment]
Me too I am experiencing the exact same single problem as yours.

**Comment by Alain-T1** [comment]
The authenticator app can be configured for many different sites, each have their own auth code... make sure you enter the one for the site listed as 
**GitHub: \<yourGitHubID\>**

**Comment by BasitJawad** [comment]
In your mobile settings set the time to use Network Provided time. and Automatic Time zone. Thats it. Now try scanning the QR code and it will be ok

**Comment by Introscopia** [comment]
Same here, Firefox authenticator app doesn't work. I get in with a recovery code, re-scan the QR, input a test code and it says "error". I installed the phone app, suppose that'll have to do now...

**Comment by BasitJawad** [comment]
set your phone time-date settings to Automatic. and then scan the QR code..
